python/pyarrow/_dataset_parquet.pyx

```diff
@@ -741,7 +737,15 @@ cdef class ParquetFragmentScanOptions(FragmentScanOptions):
             thrift_string_size_limit=self.thrift_string_size_limit,
             thrift_container_size_limit=self.thrift_container_size_limit,
         )
-        return type(self)._reconstruct, (kwargs,)
+        return _reconstruct_parquet_fragment_scan_options, (kwargs,)
+
+
+def _reconstruct_parquet_fragment_scan_options(kwargs):
+    # __reduce__ doesn't allow passing named arguments directly to the
+    # reconstructor, hence this wrapper.
+    # In Cython >= 3.0.0, function binding is turned on by default, so
+    # a global static method is used (instead of a class method) for pickling.
+    return ParquetFragmentScanOptions(**kwargs)
```

Comment:
Can we check that this change still works with Cython < 3?

Reply:
That reminds me, I wanted to add a Cython<3 nightly job to CI to test future changes don't break anything.

Reply:
BTW, I did confirm this works on Cython 0.29.36